add fig directive doc (#95)
ExecutableBookProject/cli: docs/use/myst.md

> For example, `` {numref}`my_figure` `` will produce a reference like:

Reviewer: Maybe these should be in their own short section like "referencing figures" where we can explain both `ref` and `numref`.

{numref}`my_figure`.

### Figure scaling and alignment

> Figures can also be aligned by using the option `:align: right` or `:align: left`. By default, figures are aligned to the center (see {numref}`my_figure`).

Reviewer 2: Should this be "(see Figure {numref}`my_figure`)" ?

Reviewer: nope, by default it includes `Fig.` and in the sphinx configuration there's an option for how you want that formatting to work. (e.g. `Fig` vs.`Figure`)

Author: Should we include this information in the documentation as well?

Reviewer: I think it's a good idea, since many people will not be familiar with Sphinx

Author: Is it possible to set `numfig_format` under `_config.yml`? I tried the option below but got nowhere.
```
numfig_format : {'figure': 'Figure %s', 'table': 'Table %s',
                 'code-block': 'Listing %s', 'section': 'Section %s'}
```

Author: Also, do you happen to know the difference between `%s` and `{number}`? I thought at first that `number` was a figure directive option similar to `name`, which can also be referenced as `{name}`.  Now, I'm not so sure.

````
```{figure} ../images/C-3PO_droid.png
---
scale: 25%
align: left
---
Here is my figure caption!
```
````

```{figure} ../images/C-3PO_droid.png
---
scale: 25%
align: left
---
Here is my figure caption!
```

````
```{figure} ../images/C-3PO_droid.png
---
scale: 25
align: right
---
Here is my figure caption!
```
````

```{figure} ../images/C-3PO_droid.png
---
scale: 25
align: right

---
Here is my figure caption!
```

### Figure Parameters

The following options are supported:

* `scale` : _integer percentage_

>   The scale paramater helps to uniformly scale the figure. The default is "100" which indicates no scaling. The symbol "%" is optional

Reviewer: ```suggestion
  Uniformly scale the figure. The default is "100" which indicates no scaling. The symbol "%" is optional
```

* `width` : _length or percentage_
  You can set the figure width in the following units: "em","ex","px","in","cm","mm","pt","pc","%"

* `height` : _length_
  You can set the figure height in the following units: "em","ex","px","in","cm","mm","pt","pc",""

* `alt` : _text_

>   This paramater is useful to users who might not be able to see the figure and it generally entails a short description of the figure

Reviewer: ```suggestion
  Text to be displayed if the figure cannot display, or if the reader is using assistive technologies. Generally entails a short description of the figure
```

* `align` : _"left", "center", or "right"_

>   Default alignment is center

Reviewer: ```suggestion
  Align the figure left, center, or right. Default alignment is center.
```

> * `name` : _text_

Reviewer: ```suggestion
* `name` : _text_
   A unique identifier for your figure that you can use to reference it with `{ref}` or `{numref}`. Cannot contain spaces or special characters.
```

> * `class` : _text_

Reviewer: should this be `figclass` instead of `class`? I'm not sure I know the difference between them but `figclass` is what I've seen elsewhere.

Author: I think you're right but I have to double check. `figclass` is used for figure directives while `class` is used for image directives. I've seen `figclass` in the context of aligning figures, e.g.: `:figclass: align-center`. I'll check it out and report tomorrow.

```
````

```{figure} ../images/cool.jpg
---
scale: 50%
align: right

---
Here is my figure caption!
```
### Figure Parameters

The following options are supported:

* `scale` : _integer percentage_

  Uniformly scale the figure. The default is "100" which indicates no scaling. The symbol "%" is optional.

* `width` : _length or percentage_

  You can set the figure width in the following units: "em", "ex", "px","in" ,"cm", "mm", "pt", "pc", "%".

* `height` : _length_

  You can set the figure height in the following units: "em", "ex", "px", "in", "cm", "mm", "pt", "pc", "".

* `alt` : _text_

  Text to be displayed if the figure cannot display, or if the reader is using assistive technologies. Generally entails a short description of the figure.

* `align` : _"left", "center", or "right"_

  Align the figure left, center, or right. Default alignment is center.

* `name` : _text_

  A unique identifier for your figure that you can use to reference it with `{ref}` or `{numref}`. Cannot contain spaces or special characters.

* `figclass` : _text_

  Value of the figure's class attribute which can be used to add custom CSS or JavaScript. Predefined options include:

>   * _"sidebar"_ : Display figure under sidebar

Reviewer: note - these should now be "margin" and "margin-caption". Sorry about the confusion there, this was just changed in sphinx-book-theme

Author: They'll still rendered under sidebar, correct? I'm wondering if it makes sense to change "Display figure under __sidebar__" to "Display figure under __margin__".

Reviewer: yep, agreed - in general the white space to the right of the text we are now referring to as "margin" instead of sidebar (to avoid confusion with other UI elements)

Author: sounds good!
